packages/peregrine/lib/talons/CheckoutPage/__tests__/useCheckoutPage.spec.js

```diff
@@ -0,0 +1,209 @@
+        ])
+}));
+
+jest.mock('../../../Apollo/clearCartDataFromCache', () => ({
+    clearCartDataFromCache: jest.fn()
+}));
+
+/**
+ * Constants
+ */
+
+const createCartMutation = 'createCartMutation';
+const placeOrderMutation = 'placeOrderMutation';
+const getCheckoutDetailsQuery = 'getCheckoutDetailsQuery';
+const getOrderDetailsQuery = 'getOrderDetailsQuery';
+
+const props = {
+    mutations: { createCartMutation, placeOrderMutation },
+    queries: { getCheckoutDetailsQuery, getOrderDetailsQuery }
+};
+
+const readQuery = jest.fn();
+const writeQuery = jest.fn();
+const client = { readQuery, writeQuery };
+
+const getOrderDetails = jest.fn();
+const getOrderDetailsQueryResult = jest.fn().mockReturnValue([
+    getOrderDetails,
+    {
+        data: null,
+        loading: false,
+        called: false
+    }
+]);
+const getCheckoutDetails = jest.fn();
+const getCheckoutDetailsQueryResult = jest.fn().mockReturnValue([
+    getCheckoutDetails,
+    {
+        data: null,
+        loading: false,
+        called: false
+    }
+]);
+
+const createCart = jest.fn();
+const createCartMutationResult = jest.fn().mockReturnValue([
+    createCart,
+    {
+        error: null,
+        loading: false,
+        data: null
+    }
+]);
+const placeOrder = jest.fn();
+const placeOrderMutationResult = jest.fn().mockReturnValue([
+    placeOrder,
+    {
+        error: null,
+        loading: false,
+        data: null
+    }
+]);
+
+/**
+ * Helpers
+ */
+
+const Component = props => {
+    const talonProps = useCheckoutPage(props);
+
+    return <i talonProps={talonProps} />;
+};
+
+const getTalonProps = props => {
+    const tree = createTestInstance(<Component {...props} />);
+    const { root } = tree;
+    const { talonProps } = root.findByType('i').props;
+
+    const update = newProps => {
+        act(() => {
+            tree.update(<Component {...{ ...props, ...newProps }} />);
+        });
+
+        return root.findByType('i').props.talonProps;
+    };
+
+    return { talonProps, tree, update };
+};
+
+/**
+ * beforeAll
+ */
+
+beforeAll(() => {
+    useLazyQuery.mockImplementation(query => {
+        if (query === getCheckoutDetailsQuery) {
+            return getCheckoutDetailsQueryResult();
+        } else if (query === getOrderDetailsQuery) {
+            return getOrderDetailsQueryResult();
+        } else {
+            return [jest.fn(), {}];
+        }
+    });
+
+    useMutation.mockImplementation(mutation => {
+        if (mutation === createCartMutation) {
+            return createCartMutationResult();
+        } else if (mutation === placeOrderMutation) {
+            return placeOrderMutationResult();
+        } else {
+            return [jest.fn(), {}];
+        }
+    });
+
+    useApolloClient.mockReturnValue(client);
+});
+
+/**
+ * Tests
+ */
+
+test('Should return correct shape', () => {
+    const { talonProps } = getTalonProps(props);
+
+    expect(talonProps).toMatchSnapshot();
```

Comment:
I haven't been using snapshots for these, for what it's worth.

Here's an example: https://github.com/magento/pwa-studio/blob/develop/packages/peregrine/lib/talons/CheckoutPage/__tests__/useShippingMethod.spec.js#L105-L120.

That said there are plenty examples of both approaches so no worries if you don't want to change this.
I just think the snapshot requires more human effort and is less explicit.

Reply:
I do like that approach @supernova-at. Thanks for recommending it. It is quite explicit about the shapes of the returned props. It does save us time and effort of not updating the snapshots every now and then.

But, in this component, there are a lot of props being returned and I am kind of leaning towards snapshots than explicit type checks in the interest of protecting my laziness 😉.